### PR TITLE
drivers: usbhost_hub: usb communication memory should use DRVR_ALLOC

### PR DESCRIPTION
**Pull request by chenzhihong007**
## Summary

DRVR_CTRLIN buffer must have been allocated using DRVR_ALLOC.  For example, some mcu memory should align cache line.

## Impact
only usb host hub.

### CODE REVIEW

**Review by jerpelea — APPROVED** [review]

**Comment by acassis** [comment]
MacOS is breaking again

File: drivers/usbhost/usbhost_hub.c [hunk 1]
```diff
@@ -538,42 +539,51 @@ static inline int usbhost_hubdesc(FAR struct usbhost_class_s *hubclass)
...
+      uerr("ERROR: DRVR_ALLOC failed: %d\n", ret);
+      return ret;
+    }
+
   ret = DRVR_CTRLIN(hport->drvr, hport->ep0,
-                    ctrlreq, (FAR uint8_t *)&hubdesc);
+                    ctrlreq, (FAR uint8_t *)hubdesc);
   if (ret < 0)
     {
       uerr("ERROR: Failed to read hub descriptor: %d\n", ret);
```
Comment (xiaoxiang781216) [inline drivers/usbhost/usbhost_hub.c:553]: free memory

Comment (chenzhihong007) [reply drivers/usbhost/usbhost_hub.c:553]: > free memory

line 586, free hubdesc.

Comment (xiaoxiang781216) [reply drivers/usbhost/usbhost_hub.c:553]: but you return at 554

Comment (chenzhihong007) [reply drivers/usbhost/usbhost_hub.c:554]: > but you return at 554

Exactly, I add hubdesc free if DRVR_CTRLIN failed.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**